dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -608,29 +602,46 @@ private static void TrackUniqueParameterType(ref bool hasParameterType, MethodIn
                         return null!;
                     }
 
-                    return Enum.Parse(targetType, input, ignoreCase: true);
-                };
-            }
+                    // Issue #1. Quick fix to prevent the exception thrown by ConvertTo method - System.ArgumentOutOfRangeException : Value cannot be null. (Parameter 'destinationType') (Parameter 'actual')
+                    // Context: UT - KernelFunctionTests2.ItSupportsArgumentsAsIsWithoutConvertingTheirTypeAsync, Line - await AssertParameterType<string?>(null);
+                    // For some reason, the target type that is "string?" in the method signature comes here as a string, and the above statement - 'if (wasNullable && input is null)' is not working.
```

Comment:
From the runtime's perspective, there's zero difference between `string?` and `string`: nullable reference types is a compile-time only behavior of the C# compiler. `wasNullable` is only being set to true for nullable value types, i.e. `Nullable<T>`.

This marshaling logic should not be concerned about differences between `string` and `string?`: from its perspective, they're both just `string`. Why is this extra check necessary?  If we really want to always return null if the input is null, we should remove the previous if block, as with this check we're now returning null whether or not wasNullable was true.

Reply:
Understood. The functionality has been updated to return null only if the argument is null and the target type is either a ReferenceType or a nullable ValueType. I believe we do not want to always return null when the argument is null to avoid cases where a method that accepts, for example, a parameter of type int is called with null. In this case, I think we need to throw an exception to indicate that the parameter is out of range.